pkg/vcdsdk/vapp.go

```diff
@@ -421,16 +419,22 @@ func (vdc *VdcManager) getTaskFromResponse(resp *http.Response) (*govcd.Task, er
 	return task, nil
 }
 
-func (vdc *VdcManager) SetVmExtraConfigKeyValue(vm *govcd.VM, key string, value string, required bool) error {
+func (vdc *VdcManager) SetMultiVmExtraConfigKeyValuePairs(vm *govcd.VM, extraConfigMap map[string]string,
+	required bool) (govcd.Task, error) {
 	_, extraConfigVm, err := vdc.getVmExtraConfigs(vm)
 	if err != nil {
-		return fmt.Errorf("error retrieving vm extra configs: [%v]", err)
+		return govcd.Task{}, fmt.Errorf("error retrieving vm extra configs: [%v]", err)
```

Comment:
Wouldn't sending nil be better?

Is there an easy way for the caller to know that this is an empty task object?

Reply:
I am following the pattern in govcd code. I also found this as odd. However it is better to follow the pattern IMO.